xwiki-platform-core/xwiki-platform-oldcore/src/main/java/org/xwiki/store/TemporaryAttachmentSessionsManager.java

```diff
@@ -57,6 +57,26 @@
     XWikiAttachment uploadAttachment(DocumentReference documentReference, Part part)
         throws TemporaryAttachmentException;
 
+    /**
+     * Temporary store the given {@link Part} to a cached {@link XWikiAttachment} attached to the given
+     * {@link DocumentReference}.
+     *
+     * @param documentReference the reference of the document that the attachment should be attached to
+     * @param part the actual data that is uploaded
+     * @param filename an optional filename used instead of using {@link Part#getSubmittedFileName()}, ignored when
+     *     {@code null}
+     * @return an attachment that is not saved yet but cached and contains the data of the given part
+     * @throws TemporaryAttachmentException if the part size exceeds the maximum upload size, or in case of problem
+     *     when reading the part
+     * @since 14.9RC1
+     */
+    @Unstable
+    default XWikiAttachment uploadAttachment(DocumentReference documentReference, Part part, String filename)
+        throws TemporaryAttachmentException
+    {
+        throw new UnsupportedOperationException();
```

Comment:
This basically makes the default implementation totally useless.

Reply:
Sometimes, it's possible to return something that makes sense for new methods, the only two other options I have in mind are:
- do nothing
- call `uploadAttachment(DocumentReference documentReference, Part part)

Both of which are not acceptable, so it seems like the best option is to effectively warn the user that the called implementation does not provide an implementation for the new method.

Reply:
Why not just providing the new method without making it a default and with a revapi documentation?

Reply:
Because that's not what we documented (see https://dev.xwiki.org/xwiki/bin/view/Community/CodeStyle/JavaCodeStyle/#HMakesureyourcodeisbackwardcompatible).

Also, I'm not sure the runtime error would be better with this option.

Reply:
My point is that implementing that method just to throw an exception indicating that this method is not implemented does not make any kind of sense. All this false implementation does is hiding the fact that you add a breaking change.

> Both of which are not acceptable

What's important is knowing which is worst between a failing call or one of those options.

Reply:
Updated according to the new best practices, see:
- https://github.com/xwiki/xwiki-platform/pull/1930/files#diff-2f81214bd08060b9f2f1bdd7cb0dda0e7109ccadbf852cda6ddb864abc29c8e3R139
- https://github.com/xwiki/xwiki-platform/pull/1930/commits/7b74a0986a17a8c81717a11c7212b00a2a29136f#diff-9c6fd1ffb45cadd241e751aadc5b6944a03a59ee8994fafe52c6d9e8c4707ff3R74